spring-expression/src/main/java/org/springframework/expression/spel/CodeFlow.java

```diff
@@ -346,74 +344,50 @@ public static void insertUnboxNumberInsns(
 	 * @param stackDescriptor the descriptor of the operand on top of the stack
 	 */
 	public static void insertAnyNecessaryTypeConversionBytecodes(MethodVisitor mv, char targetDescriptor, String stackDescriptor) {
-		if (CodeFlow.isPrimitive(stackDescriptor)) {
-			char stackTop = stackDescriptor.charAt(0);
-			if (stackTop == 'I' || stackTop == 'B' || stackTop == 'S' || stackTop == 'C') {
-				if (targetDescriptor == 'D') {
-					mv.visitInsn(I2D);
-				}
-				else if (targetDescriptor == 'F') {
-					mv.visitInsn(I2F);
-				}
-				else if (targetDescriptor == 'J') {
-					mv.visitInsn(I2L);
-				}
-				else if (targetDescriptor == 'I') {
-					// nop
-				}
-				else {
-					throw new IllegalStateException("Cannot get from " + stackTop + " to " + targetDescriptor);
+		if (!CodeFlow.isPrimitive(stackDescriptor)) return;
+		char stackTop = stackDescriptor.charAt(0);
+		switch (stackTop){
+			case 'I', 'B', 'S', 'C' -> {
+				switch (targetDescriptor) {
+					case 'D' -> mv.visitInsn(I2D);
+					case 'F' -> mv.visitInsn(I2F);
+					case 'J' -> mv.visitInsn(I2L);
+					case 'I' -> { //nop
+					}
+					default -> throw new IllegalStateException("Unexpected value: " + targetDescriptor);
```

Comment:
The stackTop information is no longer present in the exception message.
```suggestion
					default -> throw new IllegalStateException("Cannot get from " + stackTop + " to " + targetDescriptor);
```

Reply:
Thanks! Nice catch!

Reply:
Glad to be of help!